File: WebAPI/PurpleShop/PurpleShop.Entities/Concrete/Product.cs
using PurpleShop.Core.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace PurpleShop.Entities.Concrete
{
    public class Product : IEntity
    {
        public Product()
        {
            ProductImages = new List<ProductImage>();
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int CategoryId { get; set; }
        public virtual Category Category { get; set; }
        public int FeatureId { get; set; }
        public Feature Feature { get; set; }
        public List<ProductImage> ProductImages { get; set; }
    }
}

File: Angular/PurpleShopping/src/app/models/Product.ts
import { Category } from './Category';
import { Feature } from './Feature';
import { ProductImage } from './ProductImage';

export class Product {
  id: number;
  name: string;
  description: string;
  categoryId: number;
  category: Category;
  featureId: number;
  feature: Feature;
  productImages: ProductImage[];
}
